fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1405,7 +1390,9 @@ def _pre_backward_hook(*unused: Any) -> None:
             # Note, both ``self._rebuild_full_params`` and ``self._use_full_params`` are
             # idempotent.  So in case they are called unnecessarily, they don't incur much
             # overhead.
-            if self.ssd_offload or self.reshard_after_forward:
+            # PJ TODO: Revisit if this is still needed for ssd_offload
+            # if self.ssd_offload or self.reshard_after_forward:
+            if self.reshard_after_forward:
```

Comment:
I had always set it to delete the params in memory in the previous implementation. In the new one perhaps we don't need it. I remember fixing a particular test using this condition. Lets remove the self.ssd_offload for now and see what happens. Will try to remember what i fixed in the meantime.